tests/parser/test_class2D.py

```diff
@@ -47,53 +56,80 @@ def test_all_keys_are_different(input):
 def test_counter(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    class2d_object._count_all(class_numbers)
+    for i in range(len(class_numbers)):
+        class2d_object._count_all(class_numbers[i])
 
 
 def test_percentage(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    percentage = class2d_object.percent_all_particles_per_class(class_numbers)
-    print("Percent of particles from all data in each class:", percentage)
-    print(
-        "Percent of all particles included in the 20 most populated classes:",
-        sum(x[1] for x in percentage),
-    )
-    assert percentage[0][1] == 18.327067669172934
+    job_numbers = class2d_object.job_number
+    percentage = None
+    for i in range(len(job_numbers)):
+        percentage = class2d_object.percent_all_particles_per_class(class_numbers[i])
+        print("Percent of particles from all data in each class:", percentage)
+        print(
+            "Percent of all particles included in the 20 most populated classes for",
+            job_numbers[i],
+            ":",
+            sum(x[1] for x in percentage),
+            "%",
+        )
+    assert percentage[0][1] == 19.9430499894537
 
 
 def test_top_twenty_list(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    twenty_list = class2d_object.top_twenty_most_populated(class_numbers)
-    print("20 most populated classes:", twenty_list)
+    twenty_list = []
+    for i in range(len(class_numbers)):
+        twenty_list = class2d_object.top_twenty_most_populated(class_numbers[i])
+        print("20 most populated classes:", twenty_list)
     assert len(twenty_list) == 20
-    assert twenty_list[0][0] == "16"
+    assert twenty_list[0][0] == "16"  # this is the second list
 
 
 def test_twenty_sum(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    total = class2d_object._sum_top_twenty_particles(class_numbers)
-    assert total == 8061
+    job_numbers = class2d_object.job_number
+    total = []
+    for i in range(len(job_numbers)):
+        total.append(class2d_object._sum_top_twenty_particles(class_numbers[i]))
+        print(job_numbers[i], total[i], "particles")
+    assert sum(total) == 8599
 
 
 def test_sum_all(input):
     class2d_object = input
     class_numbers = class2d_object.class_number
-    total = class2d_object._sum_all_particles(class_numbers)
-    assert total == 10640
+    total = []
+    for i in range(len(class_numbers)):
+        total.append(class2d_object._sum_all_particles(class_numbers[i]))
```

Comment:
the same `.values()`/list bit, but also possibly a list comprehension here?
```suggestion
    total = [class2d_object._sum_all_particles(class) for class in class_numbers.values()]
```